### ISSUE TITLE
Bug: CommandMap prefix ignores override

### ISSUE THREAD

**Issue by sbonami** [capistrano/sshkit#299]
When defining a command_map override, output matches intended functionality:

    SSHKit.config.command_map[:rake] = 'pitch_fork'
    SSHKit.config.command_map[:rake] => 'pitch_fork'

However, when a prefix for the command exists, the original command is served with the prefix rather than the override:

    SSHKit.config.command_map[:rake] = 'pitch_fork'
    SSHKit.config.command_map[:rake].push("bundle exec")
    SSHKit.config.command_map[:rake] => 'bundle exec rake'

I would figure we'd get this:

    SSHKit.config.command_map[:rake] = 'pitch_fork'
    SSHKit.config.command_map[:rake].push("bundle exec")
    SSHKit.config.command_map[:rake] => `bundle exec pitch_fork`

Should L45 be something like this?:

    "#{prefixes} #{@map[command] || command}"

Related source: https://github.com/capistrano/sshkit/blob/master/lib/sshkit/command_map.rb#L42-L45

**Comment by dgtized** [comment]
:+1: I also encountered this. Unfortunately changing the code the way you suggest breaks the existing tests, which suggests that someone may be using the existing defaults logic.

I was trying to use it to swap out rake to always run a command before the generated command. Example:

```
SSHKit.config.command_map[:rake] = "rake stdout_logging"
execute :rake, "foo"
# expecting that to run
# => bundle exec rake stdout_logging foo
# but instead it runs
# => bundle exec /usr/bin/env rake
```

**Comment by mattbrictson** [comment]
It looks like this issue might be addressed in PR #311.
